File: RegexLFA/RgxNumero.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;
using System.Threading.Tasks;

namespace RegexLFA;
#pragma warning disable CS8604

internal class RgxNumero
{
    public static void GetRgxNumero()
    {
        Console.Write("Informe um valor:  ");
        var caracteres = Console.ReadLine();

        bool test = Regex.IsMatch(caracteres, "^[0-9]+$");

        if (!test)
            Console.WriteLine("O valor informado não é um numérico válido.");
        else
            Console.WriteLine("O valor informado é válido.");
    }
}

File: RegexLFA.VBNET/RgxNumero.vb
Imports System.Text.RegularExpressions

Public Class RgxNumero
    Public Shared Sub GetRgxNumero()
        Console.Write("Informe um valor:  ")
        Dim caracteres = Console.ReadLine()

        Dim test As Boolean = Regex.IsMatch(caracteres, "^[0-9]+$")

        If (test) Then
            Console.WriteLine("O valor informado é válido.")
        Else
            Console.WriteLine("O valor informado não é um numérico válido.")
        End If
    End Sub
End Class
